java/flight/flight-core/src/main/java/org/apache/arrow/flight/SessionOptionValue.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight;
+
+import java.util.Arrays;
+
+import org.apache.arrow.flight.impl.Flight;
+
+/**
+ * A union-like container interface for supported session option value types.
+ */
+public abstract class SessionOptionValue {
```

Comment:
Hmm, too bad we don't have sealed classes.

We could have a package-private (empty) constructor so that you can't subclass this from outside the package?